contracts/mainnet/connectors/fluid-vaults/main.sol

```diff
@@ -0,0 +1,299 @@
+        );
+        newColToken1_ = _adjustTokenValues(
+            getIds_[2],
+            getIds_[4],
+            newColToken1_
+        );
+        newDebtToken0_ = _adjustTokenValues(
+            getIds_[5],
+            getIds_[7],
+            newDebtToken0_
+        );
+        newDebtToken1_ = _adjustTokenValues(
+            getIds_[6],
+            getIds_[8],
+            newDebtToken1_
+        );
+
+        IVaultT4 vaultT4_ = IVaultT4(vaultAddress_);
+        IVaultT4.ConstantViews memory vaultT4Details_ = vaultT4_
+            .constantsView();
+        uint256 ethAmount_;
+
+        // Deposit token 0
+        if (newColToken0_ > 0) {
+            // Assumes ethAmount_ would either be token0 or token1
+            (ethAmount_, newColToken0_) = _handleDeposit(
+                vaultT4Details_.supplyToken.token0 == getEthAddr(),
+                newColToken0_ == type(int256).max,
+                vaultAddress_,
+                vaultT4Details_.supplyToken.token0,
+                newColToken0_
+            );
+        }
+
+        // Deposit token 1
+        if (newColToken1_ > 0) {
+            // Assumes ethAmount_ would either be token0 or token1
+            (ethAmount_, newColToken1_) = _handleDeposit(
+                vaultT4Details_.supplyToken.token1 == getEthAddr(),
+                newColToken1_ == type(int256).max,
+                vaultAddress_,
+                vaultT4Details_.supplyToken.token1,
+                newColToken1_
+            );
+        }
+
+        // Payback token 0
+        if (newDebtToken0_ < 0) {
+            ethAmount_ = _handlePayback(
+                vaultT4Details_.borrowToken.token0 == getEthAddr(),
+                newDebtToken0_ == type(int256).min,
+                vaultT4Details_.borrowToken.token0,
+                repayApproveAmtToken0_,
+                newDebtToken0_,
+                vaultAddress_
+            );
+        }
+
+        // Payback token 1
+        if (newDebtToken1_ < 0) {
+            ethAmount_ = _handlePayback(
+                vaultT4Details_.borrowToken.token1 == getEthAddr(),
+                newDebtToken1_ == type(int256).min,
+                vaultT4Details_.borrowToken.token1,
+                repayApproveAmtToken1_,
+                newDebtToken1_,
+                vaultAddress_
+            );
+        }
+
+        int256 colShares;
+        int256 debtShares;
+
+        // TODO: If max withdraw, can we send min int?
+        (nftId_, colShares, debtShares) = vaultT4_.operate{value: ethAmount_}(
+            nftId_,
+            newColToken0_,
+            newColToken1_,
+            colSharesMinMax_,
+            newDebtToken0_,
+            newDebtToken1_,
+            debtSharesMinMax_,
+            address(this)
+        );
+
+        setUint(setIds_[0], nftId_);
+
+        _setIds(setIds_[1], setIds_[3], uint256(newColToken0_));
+        _setIds(setIds_[2], setIds_[4], uint256(newColToken1_)); // TODO: what if max withdraw?
```

Comment:
only operate perfect can do max withdraw & max payback,

Reply:
Got it